src/components/Button/Button.vue

```diff
@@ -0,0 +1,381 @@
+<!--
+ - @copyright Copyright (c) 2020 Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @author Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @license GNU AGPL version 3 or any later version
+ -
+ - This program is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<docs>
+
+### General description
+General purpose button component. See props for different options.
+[Use material design icons only for icons](https://www.npmjs.com/package/vue-material-design-icons) and remember to set their size to 20.
+
+### Usage
+```
+<Button
+	text="Example text"
+	color="primary"
+	aria-label="Example aria label" />
+```
+
+</docs>
+
+<template>
+	<button
+		v-ripple
+		class="button-vue"
+		v-bind="$attrs"
+		:class="buttonClassObject"
+		:aria-label="ariaLabel"
+		:disabled="disabled"
+		v-on="$listeners"
+		@keydown.enter="makeActive"
+		@keyup.enter="makeInactive"
+		@click="tabbed=false"
+		@blur="handleBlur"
+		@keyup.tab.exact="handleTabUp"
+		@keyup.shift.tab="handleTabUp">
+		<span class="button-vue__wrapper">
+			<span v-if="hasIcon" class="button-vue__icon">
```

Comment:
Is this wrapper necessary? Couldn't we just use the bare slot like in the ActionsXXX component and style it accordingly (we anyway only want material design icons in this slot):
https://github.com/nextcloud/nextcloud-vue/blob/16cf329d4c8d8b4ad99fcc4b811c624f121ce0e4/src/assets/action.scss#L92-L100

Reply:
I think it's safer this way cause at least if a non mdi icon is provided it won't break.